src/logic/tokens/store/actions/fetchTokens.ts

```diff
@@ -100,31 +55,38 @@ export const getTokenInfos = async (tokenAddress) => {
   if (!tokenAddress) {
     return null
   }
+
   const { tokens } = store.getState()
   const localToken = tokens.get(tokenAddress)
+
   // If the token is inside the store we return the store token
   if (localToken) {
     return localToken
   }
+
   // Otherwise we fetch it, save it to the store and return it
-  const tokenContract = await getHumanFriendlyToken()
-  const tokenInstance = await tokenContract.at(tokenAddress)
-  const [tokenSymbol, tokenDecimals, name] = await Promise.all([
-    tokenInstance.symbol(),
-    tokenInstance.decimals(),
-    tokenInstance.name(),
-  ])
-  const savedToken = makeToken({
+  const [tokenDecimals, tokenName, tokenSymbol] = await generateBatchRequests({
```

Comment:
I think this generateBatchRequests function will be a nightmare to correctly type. What do you think about making a wrapper of `generateBatchRequests` for fetching token details? (I may be wrong about the typing)

Reply:
We already have something like that `getERC20DecimalsAndSymbol` in `tokenHelpers.ts`

Reply:
Yup, will do.